powermon/protocols/pi18.py

```diff
@@ -95,13 +727,13 @@ def __init__(self) -> None:
         self.add_command_definitions(QUERY_COMMANDS)
         self.add_command_definitions(SETTER_COMMANDS, result_type=ResultType.ACK)
         self.remove_command_definitions(COMMANDS_TO_REMOVE)
-        self.check_definitions_count(expected=2)
+        self.check_definitions_count(expected=21) # Count of all Commands
         self.add_supported_ports([PortType.SERIAL, PortType.USB])
 
     def check_crc(self, response: str, command_definition: CommandDefinition = None):
         """ crc check, override for now """
         log.debug("check crc for %s in pi18", response)
-        if response.startswith(b"^D"):
+        if response.startswith(b"^D") or response.startswith(b"^1") or response.startswith(b"^0"):
```

Comment:
It works the same way in my test.
It's kind of strange that the protocol is not uniform here.

Reply:
lol, the protocols are a mess, same ID with different contents, inconsistencies in the protocol itself and other weirdness